src/actions.ts

```diff
@@ -95,6 +95,7 @@ export const registerEnvironmentActions = (
             async () => {
               let actionResponse;
               try {
+                env.setIgnoreLogRestartOnSelect(true);
```

Comment:
When reveal is called, it triggers [environmentsTree.onDidChangeSelection](https://github.com/env0/vscode-env0/blob/75a5d51d01231309178a6ee5eece05a6f24c7d02/src/extension.ts#L95). This then calls restartLogs with the old deployment ID. To fix this, I added a lock. This lock prevents restartLogs from being called by onDidChangeSelection. Instead, I call restartLogs here with the new deployment ID from the server. Before adding this lock, restartLogs was called twice. First by onDidChangeSelection and then again here with the new ID. The second call would stop the previous log polling. To prevent this, I introduced the lock mechanism.